tools/Dashboard/DevHome.Dashboard/Controls/WidgetControl.xaml.cs

```diff
@@ -54,6 +76,49 @@ public WidgetControl()
         ActualThemeChanged += OnActualThemeChanged;
     }
 
+    [RelayCommand]
+    private void OnLoaded()
+    {
+        _uiSettings.TextScaleFactorChanged += HandleTextScaleFactorChangedAsync;
+    }
+
+    [RelayCommand]
+    private void OnUnloaded()
+    {
+        _uiSettings.TextScaleFactorChanged -= HandleTextScaleFactorChangedAsync;
+    }
+
+    private async void HandleTextScaleFactorChangedAsync(UISettings sender, object args)
+    {
+        await Application.Current.GetService<WindowEx>().DispatcherQueue.EnqueueAsync(() =>
+        {
+            if (WidgetSource == null)
+            {
+                return;
+            }
+
+            SetScaledWidthAndHeight(sender.TextScaleFactor);
+        });
+    }
+
+    private static double GetPixelHeightFromWidgetSize(WidgetSize size)
+    {
+        return size switch
+        {
+            WidgetSize.Small => WidgetHelpers.WidgetPxHeightSmall,
+            WidgetSize.Medium => WidgetHelpers.WidgetPxHeightMedium,
+            WidgetSize.Large => WidgetHelpers.WidgetPxHeightLarge,
+            _ => 0,
+        };
+    }
+
+    private void SetScaledWidthAndHeight(double textScale)
+    {
+        HeaderHeight = new GridLength(_headerHeightUnscaled * textScale);
+        WidgetHeight = GetPixelHeightFromWidgetSize(WidgetSource.WidgetSize) * textScale;
+        WidgetWidth = WidgetHelpers.WidgetPxWidth * textScale;
+    }
+
     private async void OpenWidgetMenuAsync(object sender, RoutedEventArgs e)
     {
         if (sender as Button is Button widgetMenuButton)

```

Comment:
`sender as button widgetMenuButton` would work here, ya?

Reply:
I think this way protects against the null case. And `if (sender as Button widgetMenuButton)` doesn't work because it doesn't evaluate to a boolean.